components/eamxx/src/physics/mam/eamxx_mam_dry_deposition_process_interface.cpp

```diff
@@ -0,0 +1,430 @@
+#include "physics/mam/eamxx_mam_dry_deposition_process_interface.hpp"
+
+// Drydep functions are stored in the following hpp file
+#include <physics/mam/eamxx_mam_dry_deposition_functions.hpp>
+
+/*
+-----------------------------------------------------------------
+NOTES:
+1. Add a CIME test and multi-process tests
+2. Ensure that the submodule for MAM4xx is the main branch
+3. Read file for fractional landuse
+-----------------------------------------------------------------
+*/
+namespace scream {
+
+MAMDryDep::MAMDryDep(const ekat::Comm &comm, const ekat::ParameterList &params)
+    : AtmosphereProcess(comm, params) {
+  /* Anything that can be initialized without grid information can be
+   * initialized here. Like universal constants, mam wetscav options.
+   */
+}
+
+// ================================================================
+//  SET_GRIDS
+// ================================================================
+void MAMDryDep::set_grids(
+    const std::shared_ptr<const GridsManager> grids_manager) {
+  using namespace ekat::units;
+
+  // set grid for all the inputs and outputs
+  // use physics grid
+  grid_ = grids_manager->get_grid("Physics");
+
+  // Name of the grid
+  const auto &grid_name = grid_->name();
+
+  ncol_ = grid_->get_num_local_dofs();       // Number of columns on this rank
+  nlev_ = grid_->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 2D (2d horiz) variable
+  const FieldLayout scalar2d{{COL}, {ncol_}};
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and
+  // interfaces
+  const FieldLayout scalar3d_mid{{COL, LEV}, {ncol_, nlev_}};
+  const FieldLayout scalar3d_int{{COL, ILEV}, {ncol_, nlev_ + 1}};
+
+  // layout for 2D (ncol, pcnst)
+  constexpr int pcnst = mam4::aero_model::pcnst;
+  const FieldLayout scalar2d_pcnct =
```

Comment:
Since you use the grid to build layouts (and that's the best way, imho), you could have used it also to build the layouts above.

Reply:
Also, the name is "scalar2d", but you call `get_2d_vector_layout`. Perhaps the name should be changed, to be less confusing?